fili-core/src/main/java/com/yahoo/bard/webservice/data/metric/mappers/RowNumMapper.java

```diff
@@ -27,35 +26,30 @@
     @Override
     public ResultSet map(ResultSet resultSet) {
 
-        Schema modifiedSchema = map(resultSet.getSchema());
-        MetricColumn metricColumn = modifiedSchema.getColumn(ROW_NUM_COLUMN_NAME, MetricColumn.class);
-
-        List<Result> newResults = new ArrayList<>();
-        Result newResult;
+        ResultSetSchema schema = map(resultSet.getSchema());
+        MetricColumn column = schema.getColumn(ROW_NUM_COLUMN_NAME, MetricColumn.class).orElseThrow(
+                () -> new IllegalStateException("Unexpected missing column")
+        );
 
         int resultSetSize = resultSet.size();
+        ArrayList<Result> newResults = new ArrayList<>(resultSetSize);
         for (int i = 0; i < resultSetSize; i++) {
-            newResult = rowNumMap(resultSet.get(i), metricColumn, i);
-            if (newResult != null) {
-                newResults.add(newResult);
-            }
+            newResults.add(rowNumMap(resultSet.get(i), column, i));
         }
 
-        ResultSet newResultSet = new ResultSet(newResults, modifiedSchema);
+        ResultSet newResultSet = new ResultSet(schema, newResults);
         LOG.trace("Mapped resultSet: {} to new resultSet {}", resultSet, newResultSet);
-
         return newResultSet;
     }
 
     @Override
-    protected Result map(Result result, Schema schema) {
-        return result;
+    protected Result map(Result result, ResultSetSchema schema) {
+        throw new UnsupportedOperationException("This code should never be reached.");
```

Comment:
Because it is potentially called only from the general map function, and it isn't here.